bluebottle/payments_docdata/adapters.py

```diff
@@ -285,13 +266,40 @@ def check_payment_status(self):
         if not hasattr(response, 'payment'):
             return None
 
-        status = self.get_status_mapping(response.payment[0].authorization.status)
-
         totals = response.approximateTotals
-        if totals.totalCaptured - totals.totalChargedback - totals.totalChargedback > 0:
-            status = StatusDefinition.SETTLED
 
-        if self.payment.status <> status:
+        if int(totals.totalAcquirerApproved) + int(totals.totalAcquirerPending) + int(totals.totalShopperPending) + int(totals.totalCaptured)== 0:
+            # No payment has been authorized
+            statuses = {payment.authorization.status for payment in response.payment}
+
+            if {'NEW', 'STARTED', 'REDIRECTED_FOR_AUTHORIZATION', 'AUTHENTICATED', 'RISK_CHECK_OK'} & statuses:
+                # All these statuses belong are considered new
+                status = StatusDefinition.STARTED
+            elif statuses == {'CANCELED', }:
+                # If all of them are cancelled, the whole payment is cancelled
+                # Yes, docdata uses "CANCELED"
+                status = StatusDefinition.CANCELLED
+            else:
+                status = StatusDefinition.FAILED
+        else:
+            # We have some authorized payments
+            if int(totals.totalChargedback) == int(totals.totalRegistered):
+                # Everything is charged back
+                status = StatusDefinition.CHARGED_BACK
+            elif int(totals.totalChargedback) + int(totals.totalRefunded) == int(totals.totalRegistered):
+                # Everything is refunded (even if it was partially charged back
+                status = StatusDefinition.REFUNDED
+            elif int(totals.totalCaptured) == int(totals.totalRegistered):
+                # Everything was captured
+                status = StatusDefinition.SETTLED
+            elif int(totals.totalAcquirerApproved) + int(totals.totalAcquirerPending) + int(totals.totalShopperPending) == int(totals.totalRegistered):
+                # Everything was authorized
+                status = StatusDefinition.AUTHORIZED
+            else:
+                # Anything else (Partly captured, partly charged back, etc)
+                status = StatusDefinition.UNKNOWN
+
+        if self.payment.status != status:
```

Comment:
Maybe we should always update this, even if the status doesn't change?
Are you sure this doesn't change while status doesn't?